lambdas/src/common/config/config-service.ts

```diff
@@ -35,10 +35,10 @@ export class ConfigService {
         if (ssmParameters.length === 0) {
             throw new Error(`No client config found. Invalid client id encountered: ${clientId}`);
         }
-        const clientConfigEntries: Map<string, string> = new Map<string, string>();
-        ssmParameters.forEach((p) => {
-            clientConfigEntries.set(p.Name!.substring(p.Name!.lastIndexOf("/") + 1), p.Value!);
-        });
+
+        const clientConfigEntries = new Map(
+            ssmParameters.map((p) => [p?.Name?.substring(p?.Name?.lastIndexOf("/") + 1), p?.Value]),
```

Comment:
The previous code had `p.Name!` which was requiring that `p.Name` not be undefined. It's also in an impure function way adding data into a variable in an enclosed scope (so in theory, could handle missing values);

The new code run this as a map, so there's a locked down 1:1 relationship on the object structure, and makes the properties optional.

I think this means we could have a mapped value of `[undefined: undefined]`. I'm not 100% sure - there's a lot of `?` in there, and it's Friday afternoon!

If we're going to make a chain of optional properties, could we get a few more tests to more clearly declare what we're expecting to happen?